### PR TITLE
Update mining script & split stencils bt code version

### PR DESCRIPTION
**Pull request by FlorianDeconinck**
## Purpose

Update the miner to the newer NSYS SQL format without breaking backward compatibility & split stencil per code hash.

## Code changes:

- Templated queries
- NSYS SQL version detection (empirical)
- Introduce better handling for runs with no halo ex tagging.
- Stencil name detection doesn't obliterate the hash anymore, leading to splitting timing for different version of the code.
- Better name clean up for stencils (from GT meta-tamplated funcation name).

## Requirements changes:

N/A

## Infrastructure changes:

N/A

## Checklist
Before submitting this PR, please make sure:

- [x] You have followed the coding standards guidelines established at [Code Review Checklist](https://paper.dropbox.com/doc/Code-Review-Checklist--BD7zigBMAhMZAPkeNENeuU2UAg-IlsYffZgTwyKEylty7NhY).
- [x] Docstrings and type hints are added to new and updated routines, as appropriate
- [x] All relevant documentation has been updated or added (e.g. README, CONTRIBUTING docs)
- [x] Unit tests are added or updated for non-stencil code changes

### CODE REVIEW

File: profiler/nsys_mine.py [hunk 1]
```diff
@@ -178,14 +188,20 @@ def _plot_total_call(fv3_kernel_timings: Dict[str, List[int]]):
...
     )
     if approx_stencil_name_re is None:
         return kernels
-    approx_stencil_name = approx_stencil_name_re.group().lstrip("0123456789 ")
+    # Clean up & insert
+    approx_stencil_name = (
+        approx_stencil_name_re.groups()[0]
+        .lstrip("IN0123456789<")
+        .replace(" ", "")
+        .replace("____gtcuda", "")
```
Comment (eddie-c-davis) [inline profiler/nsys_mine.py:203]: Might be useful to parameterize the backend, could be `gtc:cuda` soon. That can wait of course...

File: profiler/nsys_mine.py [hunk 2]
```diff
@@ -258,6 +281,7 @@ def parse_args():
                 break
     assert skip_first_mainloop is False and min_start != 0
     timestep_time_in_ms = (float(max_end) - float(min_start)) * 1.0e3
+    print(f"Mining timespte between {min_start} and {max_end}")
```
Comment (eddie-c-davis) [inline profiler/nsys_mine.py:284]: Potential typo, `timespte`, should be `timestep`?

File: profiler/nsys_mine.py [hunk 3]
```diff
@@ -350,10 +385,26 @@ def parse_args():
     percentage_of_python_overhead_time = (
         (timestep_time_in_ms - total_gpu_kernel_time_in_ms) / timestep_time_in_ms
     ) * 100.0
-    percentage_of_python_without_haloex_overhead_time = (
-        (timestep_time_in_ms - total_gpu_kernel_time_in_ms - halo_ex_time_in_ms)
-        / timestep_time_in_ms
-    ) * 100.0
+    if halo_tag_found:
```
Comment (eddie-c-davis) [inline profiler/nsys_mine.py:388]: Nice improvement!

**Review by eddie-c-davis — APPROVED** [review]
Looks great!  Couple of minor comments...

File: profiler/nsys_mine.py (hunk 1 above)
Comment (FlorianDeconinck) [reply profiler/nsys_mine.py:203]: Good point but I expect the generated code to differ so much I'll need to parametrize this entire routine

File: profiler/nsys_mine.py (hunk 2 above)
Comment (FlorianDeconinck) [reply profiler/nsys_mine.py:284]: Nice catch

**Pull request merged**